goblint_runner.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+import argparse
+from dataclasses import dataclass
+
+import subprocess
+from os import path
+import logging
+
+class GoblintRunner:
+
+    def __init__(self, logger):
+        self.logger = logger
+        self.current_path = path.dirname(__file__)
+        self.goblint_executable_path = path.join(self.current_path, "goblint")
+        if not path.exists(self.goblint_executable_path):
+            logger.error(f" Could not find goblint executable at {self.goblint_executable_path}; did you build goblint with make?")
+            exit(1)
+
+        parser = argparse.ArgumentParser(
+            description="""A facade in front of goblint to enable running a portfolio of configurations for SV-COMP.
+            All args apart from --portfolio-conf/-p are passed on to the actual goblint calls.
+            The portfolio config file is a plaintext file whose lines each consist of goblint parameters, in particular including 
+            --conf followed by a path to a goblint config file (relative to the goblint base dir, or absolute). 
+            Goblint is run with each parameterset in order of specification as long as goblint produces an unknown verdict or reaches the end of the list.
+            You may add comments to the portfolio config file by starting a line with #.
+            """
+        )
+        parser.add_argument("-p","--portfolio-conf", type=str, metavar="FILE",dest="portfolio",
+                            help="a path to a portfolio configuration file (relative to goblint_runner.py)")
+        conf_args, self.other_args = parser.parse_known_args()
+        logger.debug(f"Portfolio-conf file: {conf_args.portfolio}")
+        logger.debug(f"Arguments passed on to goblint: {" ".join(self.other_args)}")
+
+        self.configs = []
+        if conf_args.portfolio:
+            if not path.exists(conf_args.portfolio):
+                logger.error(f" Could not find portfolio conf file at {conf_args.portfolio}")
+                exit(1)
+            with open(conf_args.portfolio, "r") as conflist_file:
+                self.configs = [c.strip() for c in conflist_file.readlines() if not c.strip().startswith("#")]
+            logger.info(f"Loaded goblint configs: {", ".join(self.configs)}")
+            
+    def run_with_config(self, config_str):
+        config_args = config_str.split(" ")
+        # in config_args, replace any relative paths with absolute paths
+        pathparameters=["--conf"] # add more if needed
+        for i in range(len(config_args)):
+            if config_args[i] in pathparameters and i + 1 < len(config_args):
+                conf_path = config_args[i + 1]
+                if not path.isabs(conf_path):
+                    abs_conf_path = path.join(self.current_path, conf_path)
+                    config_args[i + 1] = abs_conf_path
```

Comment:
This shouldn't be necessary: Goblint should already find confs relative to itself.